config/service_config.go

```diff
@@ -119,8 +119,8 @@ func getRandomPort(protocolConfigs []*ProtocolConfig) *list.List {
 		if err != nil {
 			panic(perrors.New(fmt.Sprintf("Get tcp port error,err is {%v}", err)))
 		}
-		defer tcp.Close()
 		ports.PushBack(strings.Split(tcp.Addr().String(), ":")[1])
+		tcp.Close()
```

Comment:
I am not very sure whether `tcp.Addr().String()` will return something like "127.0.0.1" which will cause `ports.PushBack(strings.Split(tcp.Addr().String(), ":")[1])` panic.

Reply:
updated.